homeassistant/components/elmax/alarm_control_panel.py

```diff
@@ -0,0 +1,111 @@
+"""Elmax sensor platform."""
+from __future__ import annotations
+
+from elmax_api.model.alarm_status import AlarmArmStatus, AlarmStatus
+from elmax_api.model.command import AreaCommand
+from elmax_api.model.panel import PanelStatus
+
+from homeassistant.components.alarm_control_panel import (
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityFeature,
+    CodeFormat,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_ALARM_ARMED_AWAY, STATE_ALARM_DISARMED
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import InvalidStateError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElmaxCoordinator
+from .common import ElmaxEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Elmax area platform."""
+    coordinator: ElmaxCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    known_devices = set()
+
+    def _discover_new_devices():
+        panel_status: PanelStatus = coordinator.data
+        # In case the panel is offline, its status will be None. In that case, simply do nothing
+        if panel_status is None:
+            return
+
+        # Otherwise, add all the entities we found
+        entities = []
+        for area in panel_status.areas:
+            # Skip already handled devices
+            if area.endpoint_id in known_devices:
+                continue
+            entity = ElmaxArea(
+                panel=coordinator.panel_entry,
+                elmax_device=area,
+                panel_version=panel_status.release,
+                coordinator=coordinator,
+            )
+            entities.append(entity)
+
+        if entities:
+            async_add_entities(entities)
+            known_devices.update([e.unique_id for e in entities])
+
+    # Register a listener for the discovery of new devices
+    remove_handle = coordinator.async_add_listener(_discover_new_devices)
+    config_entry.async_on_unload(remove_handle)
```

Comment:
You could condense this into one expression if you wanted to.

Reply:
Done, hope it's better now.